autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnResponseUtils.java

```diff
@@ -17,25 +18,34 @@ public class YarnResponseUtils {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(YarnResponseUtils.class);
 
-    public List<String> getYarnRecommendedDecommissionHostsForHostGroup(String clusterCrn, YarnScalingServiceV1Response yarnResponse,
-            Map<String, String> hostFqdnsToInstanceId, int maxAllowedDownScale, Optional<Integer> mandatoryDownScaleCount, Integer maxScaleDownStepSize) {
+    public List<String> getYarnRecommendedDecommissionHostsForHostGroup(YarnScalingServiceV1Response yarnResponse,
+            Map<String, String> hostFqdnsToInstanceId, List<String> stoppedInstanceIds, int maxAllowedDownScale, Optional<Integer> mandatoryDownScaleCount,
+            Optional<Integer> mandatoryStopStartDownscaleCount, Integer maxScaleDownStepSize) {
         // yarnResponse may not have maxAllowedDownscale candidates. This is expected when using LoadBasedDownscaling
         // where a force downscale is not required. In this case downscaling is done based on yarn recommended count.
-        Integer allowedDownscale = mandatoryDownScaleCount.orElse(maxAllowedDownScale);
+        Integer allowedDownscale = mandatoryStopStartDownscaleCount.orElse(mandatoryDownScaleCount.orElse(maxAllowedDownScale));
```

Comment:
If mandatoryStopStartDownscaleCount always wins - is there a possibility of taking the cluster below the minCount of HEALTHY hosts ?
What happens when there are UNHEALTHY hosts on the cluster ? Should these be downscale candidates / delete candidates ? (This is a tough question given we don't know why these nodes are marked UNHEALTHY - and they could still be running workloads)

Reply:
> What happens when there are UNHEALTHY hosts on the cluster ? Should these be downscale candidates / delete candidates ?

Tested this scenario out. UNHEALTHY instances _can_ be chosen as mandatory downscale if they are running workloads, in combination with any stopped instances available on the cluster.  

> If mandatoryStopStartDownscaleCount always wins - is there a possibility of taking the cluster below the minCount of HEALTHY hosts ?

Since mandatoryDownscaleAdjustment is a measure of the offset between the total hostGroup nodeCount (`existingHostGroupSize`) and the `maxResourceValue` specified in the AS policy, it shouldn't go below the lower bound but only reach the given upper bound.